crates/ruff_db/src/parsed.rs

```diff
@@ -0,0 +1,99 @@
+use std::fmt::Formatter;
+use std::ops::Deref;
+use std::sync::Arc;
+
+use ruff_python_ast::{ModModule, PySourceType};
+use ruff_python_parser::{parse_unchecked_source, Parsed};
+
+use crate::source::source_text;
+use crate::vfs::{VfsFile, VfsPath};
+use crate::Db;
+
+/// Returns the parsed AST of `file`, including its token stream.
+///
+/// The query uses Ruff's error resilient parser. That means that the parser always succeeds to produce a
+/// AST even if the file contains syntax errors. The syntax errors are Parsing the module succeeds even when the file contains syntax error. The parse errors
+/// are then accessible through [`Parsed::errors`].
+///
+/// The parse tree is cached between invocations, but the query doesn't make use of Salsa's optimization
```

Comment:
I found this sentence confusing on first read. Clearer would be:
```suggestion
/// Salsa caches the parse tree between invocations if the `VfsFile` hasn't changed, but the query
/// doesn't make use of Salsa's additional optimization
```